src/repositories/docsetsRepository.ts

```diff
@@ -0,0 +1,119 @@
+import { Db } from 'mongodb';
+import { BaseRepository } from './baseRepository';
+import { ILogger } from '../services/logger';
+import { IConfig } from 'config';
+
+const docsetsCollectionName = process.env.DOCSETS_COL_NAME || 'docsets';
+export class DocsetsRepository extends BaseRepository {
+  constructor(db: Db, config: IConfig, logger: ILogger) {
+    super(config, logger, 'DocsetsRepository', db.collection(docsetsCollectionName));
+  }
+
+  private getAggregationPipeline(
+    matchConditionField: string,
+    matchConditionValue: string,
+    projection?: { [k: string]: number }
+  ) {
+    return [
+      // Stage 1: Unwind the repos array to create multiple documents for each referenced repo
+      {
+        $unwind: '$repos',
+      },
+      // Stage 2: Lookup to join with the repos_branches collection
+      {
+        $lookup: {
+          from: 'repos_branches',
+          localField: 'repos',
+          foreignField: '_id',
+          as: 'repo',
+        },
+      },
+      // Stage 3: Match documents based on given field
+      {
+        $match: {
+          [`repo.${matchConditionField}`]: matchConditionValue,
+        },
+      },
+      // Stage 4: Merge/flatten repo into docset
+      {
+        $replaceRoot: { newRoot: { $mergeObjects: [{ $arrayElemAt: ['$repo', 0] }, '$$ROOT'] } },
+      },
+      // Stage 5: Exclude fields
+      {
+        $project: projection || {
+          _id: 0,
+          repos: 0,
+          repo: 0,
+        },
+      },
+    ];
+  }
+
+  async getProjectByRepoName(repoName: string) {
```

Comment:
would prefer return types in these functions. Promise<string>

Reply:
I'll work on that! These were straight from repoBranchesRepository.